MEDimage/learning/DesignExperiment.py

```diff
@@ -0,0 +1,454 @@
+                    combs = [comb for i in range(len(vars)) for comb in combinations(var_names, i+1)]
+                    combstrings = ['_'.join(elt) for elt in combs]
+                    ml['variables']['combinations'] = combstrings
+                else:
+                    ml['variables']['combinations'] = var_options['combinations']
+            
+            # Varibles to use for ML
+            ml['variables']['varStudy'] = var_options['varStudy']
+
+        # ML algorithms
+        if 'algorithms' in options:
+            algorithm = ml['settings']['algorithm']
+            algorithms = load_json(ml_options['algorithms'])
+            ml['algorithms'] = {}
+            ml['algorithms'][algorithm] = algorithms[algorithm]
+            flag_feat_set_selection = False
+            if 'fset_selection' in ml['algorithms'][algorithm].keys():
+                flag_feat_set_selection = True
+                feat_set_selection = ml['algorithms'][algorithm]['fset_selection']['method']
+
+        # ML data processing methods and its options
+        for (method, method_list) in [('datacleaning', all_datacleaning), ('imbalance',all_imbalance),
+                                    ('normalization', all_normalization), ('fSetReduction', all_fset_reduction)]:
+            # Skip if no method is selected
+            if all(v == "" for v in method_list):
+                continue
+            if method in options:
+                # Add algorithm specific methods
+                if method in ml['settings'].keys():
+                    method_list.append(ml['settings'][method])
+                    method_list = list(set(method_list))  # to only get unique values of all_datacleaning
+                method_options = load_json(ml_options[method]) # load json file of each method
+                if method == 'normalization' and 'combat' in method_list:
+                    ml[method] = 'combat'
+                    continue
+                ml[method] = dict()
+                for name in list(set(method_list)):
+                    if name != "":
+                        ml[method][name] = method_options[name]
+
+        # ML feature set selection options
+        if 'fset_selection' in options:
+            feat_set_selection_content = load_json(ml_options['fset_selection'])
+            if flag_feat_set_selection:
+                ml['fset_selection'] = dict()
+                ml['fset_selection'][feat_set_selection] = feat_set_selection_content[feat_set_selection]
+
+        # Save the ML dictionary
+        if self.experiment_label == "":
+            raise ValueError("Experiment label is empty. Class was not initialized properly.")
+        path_ml_object = self.path_study / f'ml__{self.experiment_label}.json'
+        ml = posix_to_string(ml)    # Convert all paths to string
+        save_json(path_ml_object, ml)
+        
+        # return ml
+        return path_ml_object
```

Comment:
Grosse grosse fonction. Possible de la diviser? Chacune des grandes boucles `for` pourraient aller dans une méthode différente de ta classe. Sinon c'est difficile à débugger.